redash/query_runner/__init__.py

```diff
@@ -106,6 +109,37 @@ def find_last_keyword_idx(parsed_query):
     return -1
 
 
+def get_column_types_from_dataframe(df: pd.DataFrame) -> list:
+    columns = []
+
+    for column_name, column_type in df.dtypes.items():
+        if column_type == np.bool_:
+            redash_type = TYPE_BOOLEAN
+        elif column_type == np.inexact:
+            redash_type = TYPE_FLOAT
+        elif column_type == np.integer:
+            redash_type = TYPE_INTEGER
+        elif column_type in (np.datetime64, np.dtype("<M8[ns]")):
+            if df.empty:
+                redash_type = TYPE_DATETIME
+            elif len(df[column_name].head(1).astype(str).loc[0]) > 10:
+                redash_type = TYPE_DATETIME
+            else:
+                redash_type = TYPE_DATE
+        else:
+            redash_type = TYPE_STRING
```

Comment:
Can we make this a mapping?

```python
column_type_mappings = {
  np.bool_: TYPE_BOOLEAN,
  np.inexact: TYPE_FLOAT,
  ...
}

redash_type = column_type_mappings[column_type]
```

Reply:
Oh I see, this was moved from elsewhere. I'd be fine keeping it as is then, but obviously better if it's made more readable :)

Reply:
Original code was also written by me :)

Thought about mapping here, but using extra if section for date/datetime prevented me from doing this.